tuplex/codegen/include/BlockGeneratorVisitor.h

```diff
@@ -107,15 +108,17 @@ namespace codegen {
                 //         assert(llvm::isa<llvm::Constant>(nullPtr));
                 // }
 
-                return codegen::SerializableValue(builder.CreateLoad(ptr), builder.CreateLoad(sizePtr),
+                // iterator slot may not have ptr yet
+                return codegen::SerializableValue(ptr ? builder.CreateLoad(ptr) : nullptr, builder.CreateLoad(sizePtr),
                         nullPtr ? builder.CreateLoad(nullPtr) : nullptr);
             }
 
             inline void store(llvm::IRBuilder<>& builder, const codegen::SerializableValue& val) {
-                assert(ptr && sizePtr);
+                // assert(sizePtr);
```

Comment:
same regarding it, yet I think leaving the iterator slot may not have ptr yet comment is good. 

Maybe add to description when this is the case?

Reply:
Kept the assertion.
Changed to use dummy LLVM type for iterators during declareVariables() so iterator's slot ptr won't become nullptr.
Comments were added in corresponding places.